clusterloader2/docs/experiments.md

```diff
@@ -0,0 +1,97 @@
+# Clusterloader2 - experiment rollout
+
+In this doc any change to the behavior of clusterloader2 is referred as an
+"experiment".
+
+## Motivation
+
+clusterloader2 is a tool used by all scalability, performance tests. Tests
+compile clusterloader2 at [HEAD], thus introducing breaking changes to
+clusterloader2 will stop scalability tests from passing at all. They large blast
+radius: every PR to k/k needs to pass. Also while for smaller and faster tests,
+breakages aren't that costly (unless they happen on the weekend, see
+https://github.com/kubernetes/perf-tests/pull/586), they are expensive for
+large, rarely run tests (e.g. [ci-kubernetes-e2e-gce-scale-performance]).
+
+For this reason, all new changes/features added to clusterloader2 shall be gated
+and rolled out gradually. This way, we can minimize blast radius of breaking
+changes. We can even stop some kinds of them to happen at all, as they should be
+caught at presubmit time and not allowed to merge at all.
+
+## Experiment rollout
+
+### Experiment design
+
+
+### General principles
+
+1. Allow at least 24h between changes to test configs to ensure the experiment
+   is stable. We don't want to block PR to k/k because of a flaky experiment in
+   clusterloader2.
+
+1. Check with sig-scalability whether there is a regression in the test you want
+   to enable experiment for. We don't want new features in clusterloader2 to
+   interfere with regression' debugging.
+
+1. For tests not listed below it's fine to enable experiments at your
+   convenience.
+
+### Step-by-step process
+
+_Each step should be a separate PR_
+
+1. Add a "knob" to turn on the future experiment. Usually this means adding a
+   new environmental variable (PR to [test-infra]) or a new override file (PR to
+   [perf-tests]). At this point the "knob" is not used by any code path.
+
+   Since the knob is not used anywhere it's no-op and should be safe.
+
+
+1. Enable experiment in [perf-tests presubmits] and
+   [ci-kubernetes-e2e-gci-gce-scalability]
+
+   Again cince the knob is not used anywhere it's no-op and should be safe.
+
+   Perf-test presubmit runs two jobs: [pull-perf-tests-clusterloader2] and
+   [pull-perf-tests-clusterloader2-kubemark]. Primary role of those presubmits
+   is to catch bugs in code from perf-tests, so we enable the experiment for
+   both of those jobs first. Enabling experiment in
+   [ci-kubernetes-e2e-gci-gce-scalability] should give you enough data points to
```

Comment:
It's a release blocking job, so we should avoid making any changes there during code-freeze/thaw

Reply:
Added info, that one should wait until the freeze is suspended.